packages/next/src/server/app-render/action-handler.ts

```diff
@@ -517,7 +524,10 @@ export async function handleAction({
     // This might be an old browser that doesn't send `host` header. We ignore
     // this case.
     warning = 'Missing `origin` header from a forwarded Server Actions request.'
-  } else if (!host || originDomain !== host.value) {
+  } else if (
+    !host ||
+    new RegExp('^' + host.value + '(.?)$').test(originDomain)
```

Comment:
Let's drop this for now as this doesn't seem valid and we should track down where the invalid trailing dot is coming from instead.

Reply:
Okidoki, not sure how Vercel works from the inside, so can't help on that part. Can come from some proxy or the DNS

Reply:
@jperezr21 could you share the project you are seeing this on ideally a specific deployment URL?